controller-server/src/main/java/com/yahoo/vespa/hosted/controller/Application.java

```diff
@@ -118,6 +121,20 @@ public Application(TenantAndApplicationId id, Instant createdAt, DeploymentSpec
         return versions;
     }
 
+    /** Returns the currently deployed versions of the application */
+    public Collection<ApplicationVersion> deployableVersions(boolean ascending) {
+        Deque<ApplicationVersion> versions = new ArrayDeque<>();
+        String previousHash = "";
+        for (ApplicationVersion version : versions()) {
+            if (previousHash.equals(previousHash = version.bundleHash().orElse("")) && version.bundleHash().isPresent())
```

Comment:
Consider putting assignment on separate line. I don't think I've ever seen an equals call that also assigns a variable. 😆

Reply:
I've seen it multiple places in the standard library 😬 But sure. It's just what felt natural writing. Maybe not the easiest thing to parse. though ... 😀